File: examples/zeromq_server.py
import json
import zmq

context  = zmq.Context()
socket   = context.socket(zmq.REP)
socket.bind('tcp://127.0.0.1:5555')

while True:
    meta = socket.recv()
    body = socket.recv()

    (method, path, headers) = json.loads(meta)

    socket.send(json.dumps((200, {"Content-Type": 2})), zmq.SNDMORE)
    socket.send(body)
    

File: examples/zeromq_server.rb
require 'yajl'
require 'ffi-rzmq'

context = ZMQ::Context.new
socket  = context.socket ZMQ::REP
socket.bind 'tcp://127.0.0.1:5555'

loop do
  socket.recv_string meta=''
  socket.recv_string body=''

  method, path, headers = Yajl.load(meta)

  socket.send_string Yajl.dump([
    200, {'Content-Length' => 2}]), ZMQ::SNDMORE
  socket.send_string 'ok'
end

